samples/debugmalloc/enclave/enc.c

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/debugmalloc.h>
+#include <stdio.h>
+
+// Include the trusted debugmalloc header that is generated
+// during the build. This file is generated by calling the
+// sdk tool oeedger8r against the debugmalloc.edl file.
+#include "debugmalloc_t.h"
+
+// This is the function that the host calls. It prints
+// a message in the enclave before calling back out to
+// the host to print a message from there too.
+void enclave_hello()
+{
+    void *buf1, *buf2, *buf3;
+    uint64_t num;
+    char* report;
+
+    oe_assert(oe_debug_malloc_tracking_start() == OE_OK);
+    buf1 = oe_malloc(4096);
+    oe_assert(oe_debug_malloc_tracking_stop() == OE_OK);
+    if (oe_debug_malloc_tracking_report(&num, &report) == OE_OK)
+    {
+        printf("There are %d un-freed objects reported:\n%s\n", num, report);
+        oe_free(report);
+    }
+
+    oe_assert(oe_debug_malloc_tracking_start() == OE_OK);
+    buf2 = oe_malloc(4096);
+    oe_assert(oe_debug_malloc_tracking_stop() == OE_OK);
+
+    oe_assert(oe_debug_malloc_tracking_start() == OE_OK);
+    buf3 = oe_malloc(4096);
+    oe_assert(oe_debug_malloc_tracking_stop() == OE_OK);
+    if (oe_debug_malloc_tracking_report(&num, &report) == OE_OK)
+    {
+        printf("There are %d un-freed objects reported:\n%s\n", num, report);
+        oe_free(report);
+    }
+
+    oe_free(buf1);
+    oe_free(buf2);
+    //    oe_free(buf3);
```

Comment:
Shouldn't have commented out code.  If you want to force a leak, explain it in the comment like

```suggestion
    // We intentionally don't call oe_free(buf) here, so that the sample will demonstrate leak detection.
```

Reply:
Considering samples also run in release mode without debug malloc, it is not good to ship a sample with memory leaking. I am updating the doc to let users try by themselves.